social-tw/social-tw-website
# Refactor update circuit version

## Summary

Refactor and update the circuit to `circom 2.1.0`. 

## Linked Issue

#70

## Details

There are 2 main part changed:
1. `packages/circuits/circuits/dataProof.circom` refactoring and the files built from snark.
2. `packages/circuits/package.json` adding a script (`buildOverwrite`) to rebuilt the snark files based on the new circuit. 

## Impacted Areas

The main changed file would be `packages/circuits/circuits/dataProof.circom`, with the circuit based snark file changed. (you can find the built files in `packages/circuits/zkBuild` directory) 


## Tests

The test is generated from `create-unirep-app`.
- Prove data in Unirep App
    - should generate a data proof
    - should generate a data proof with values
    - should not generate a data proof with invalid values 

## Possible Impacts


## Visual Materials

no

## Verification Steps

First, at the root dir, run
```
yarn install && yarn build
```

Which would build the files you needed for test.

After that, run
```
yarn circuits test
```
to run the tests of circuits.

The result should pass 3 tests:

```
yarn circuits test                                                                                                                  ─╯
yarn run v1.22.19
$ yarn workspace @unirep-app/circuits run test
$ mocha -r ts-node/register test/*.test.ts --exit


  Prove data in Unirep App
Gen Proof time: 1081 ms (1 s)
    ✔ should generate a data proof (1444ms)
Gen Proof time: 962 ms (0 s)
    ✔ should generate a data proof with values (1335ms)
ERROR:  4 Error in template DataProof_78 line: 45

    ✔ should not generate a data proof with invalid values (66ms)


  3 passing (3s)

✨  Done in 5.31s.
```

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [X] refactor the circuit 
-   [X] added a script to rebuild the snark-related files

## Checklist
